gql/client.py

```diff
@@ -525,7 +554,35 @@ async def _subscribe(
         finally:
             await inner_generator.aclose()
 
+    @overload
+    async def subscribe(
+        self,
+        document: DocumentNode,
+        *args,
+        variable_values: Optional[Dict[str, Any]] = ...,
+        operation_name: Optional[str] = ...,
+        serialize_variables: Optional[bool] = ...,
+        parse_result: Optional[bool] = ...,
+        get_execution_result: Literal[False] = ...,
+        **kwargs,
+    ) -> AsyncGenerator[Dict[str, Any], None]:
+        ...
+
+    @overload
     async def subscribe(
+        self,
+        document: DocumentNode,
+        *args,
+        variable_values: Optional[Dict[str, Any]] = ...,
+        operation_name: Optional[str] = ...,
+        serialize_variables: Optional[bool] = ...,
+        parse_result: Optional[bool] = ...,
+        get_execution_result: Literal[True],
+        **kwargs,
+    ) -> AsyncGenerator[ExecutionResult, None]:
+        ...
+
+    async def subscribe(  # type: ignore[misc]
```

Comment:
This `# type: ignore` directive is required because Mypy is unable to infer that `subscribe` will only ever yield a single type for a particular instantiation of the generator. There's probably a complex way to make this work properly by moving out some functionality into another overloaded function, but I worry that it might overcomplicate the code.

Reply:
It was solved by removing the async keyword in the overloaded definitions.